### PR TITLE
Clean up on Isle 1 for Resolvers + New PrivateCollectionResolver

### PR DESCRIPTION
**Pull request by wshafer**

### CODE REVIEW

File: src/Resolver/FileResolverAbstract.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public function resolveFile($file)
+    {
+        if (filter_var($file, FILTER_VALIDATE_URL)) {
+            return $this->getHtmlAsset($file);
+        }
+
+        return $this->getFileAsset($file);
+    }
+
+    protected function getHtmlAsset($filePath)
```
Comment (RWOverdijk) [inline src/Resolver/FileResolverAbstract.php]: Docblock

File: src/Resolver/FileResolverAbstract.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        return new HttpAsset($filePath);
+    }
+
+    protected function getFileAsset($filePath)
```
Comment (RWOverdijk) [inline src/Resolver/FileResolverAbstract.php]: Docblock

File: src/Resolver/PathStackResolver.php [hunk 3]
```diff
@@ -176,17 +121,14 @@ public function resolve($name)
...
 
-                $asset->mimetype = $mimeType;
-
-                return $asset;
+            if (!$asset) {
+                return null;
             }
+
+            $asset->mimetype = $this->getMimeResolver()->getMimeType($name);
+            return $asset;
```
Comment (RWOverdijk) [inline src/Resolver/PathStackResolver.php]: Newline before return

File: src/Resolver/PrivateCollectionResolver.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        if (!is_array($collectionMap[$name])) {
+            throw new Exception\RuntimeException(
+                "Collection with name $name is not an an array."
+            );
+        }
+
+        $collection = new AssetCollection;
+        $collection->setTargetPath($name);
```
Comment (RWOverdijk) [inline src/Resolver/PrivateCollectionResolver.php]: Newline between assignment and method call

File: src/Resolver/PrivateCollectionResolver.php [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        $collection->mimetype = $this->getMimeResolver()->getMimeType($name);
+
+        return $collection;
+    }
+
+    /**
+     * @param string $path
+     * @return \Assetic\Asset\FileAsset|\Assetic\Asset\HttpAsset|null
```
Comment (RWOverdijk) [inline src/Resolver/PrivateCollectionResolver.php]: Newline in between docblock types

File: src/Resolver/PrivateCollectionResolver.php [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        $asset = $this->resolveFile($path);
+
+        if (empty($asset)) {
+            throw new Exception\RuntimeException(
+                'Unable to locate file: '.$path
+            );
+        }
+
+        $this->getAssetFilterManager()->setFilters($path, $asset);
+        return $asset;
```
Comment (RWOverdijk) [inline src/Resolver/PrivateCollectionResolver.php]: Newline before return

**Review by RWOverdijk — CHANGES_REQUESTED** [review]
Quick scan, not a full review.

**Comment by wshafer** [comment]
Breaking this out into smaller pieces.

**Pull request merged**